server/routerlicious/packages/routerlicious/src/kafka-service/resourcesFactory.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { KafkaResources, KafkaResourcesFactory } from "@fluidframework/server-services-ordering-kafkanode";
+import { RdkafkaResourcesFactory } from "@fluidframework/server-services-ordering-rdkafka";
+import { IResourcesFactory } from "@fluidframework/server-services-utils";
+import { Provider } from "nconf";
+
+/**
+ * A generic kafka resources factory that picks rdkafka / kafka-node based on the config
+ */
+export class OrderingResourcesFactory implements IResourcesFactory<KafkaResources> {
```

Comment:
The RdKafkaResources factory create() method returns a Promise<RdkafkaResources>.
Should create() from this new OrderingResourceFactory return a Promise<IKafkaResources> instead?

In other words, is the KafkaResources type compatible with RdkafkaResources?

Reply:
`RdkafkaResources` is compatible with `KafkaResources`. They are actually the same interface / class but with different names. Ideally they wouldn't be separate and they should share the name resources class. However it was hard to find a shared place in the codebase to put that class. So for now they are separate.